src/leaseCreateCharge/saga.ts

```diff
@@ -30,28 +30,35 @@ function* fetchAttributesSaga(): Generator<any, any, any> {
 }
 
 function* fetchReceivableTypesSaga(): Generator<any, any, any> {
+  let nextUrl = ''
+  const allReceivableTypes = []
   try {
-    const {
-      response: {
-        status: statusCode
-      },
-      bodyAsJson
-    } = yield call(fetchReceivableTypes);
+    while (typeof nextUrl === 'string') {
```

Comment:
Is this enough of a check to see whether to continue? Maybe something else could be checked to ensure it doesn't end up in infinite loop

Reply:
If there are more pages, `bodyAsJson.next` will be a string that contains the URL. If there are no pages left, `bodyAsJson.next` will be null.

In all cases of the switch case below, the nextUrl will be either a string (should continue), or null (should stop). If there is an error in the `call(fetchRecceivableTypes, nextUrl)` line, the error is caught and nextUrl is again set to null, just in case even though we are out of the loop by then.

The condition could also be `(nextUrl !== null)`, but it should achieve the same thing.